compute_sdk/tests/unit/test_executor.py

```diff
@@ -718,42 +810,55 @@ def test_reload_starts_new_watcher(gc_executor):
     gcc.web_client.get_batch_status.return_value = mock_batch_result
 
     client_futures = list(gce.reload_tasks())
-
     assert len(client_futures) == num_tasks
-    try_assert(lambda: gce._result_watcher.is_alive())
-    watcher_1 = gce._result_watcher
+
+    watcher_1 = _RESULT_WATCHERS.get(gce.task_group_id)
+    try_assert(lambda: watcher_1.is_alive())
 
     gce.reload_tasks()
-    try_assert(lambda: gce._result_watcher.is_alive())
-    watcher_2 = gce._result_watcher
 
-    assert watcher_1 is not watcher_2
+    watcher_2 = _RESULT_WATCHERS.get(gce.task_group_id)
+    try_assert(lambda: watcher_2.is_alive())
 
+    assert watcher_1 is watcher_2
```

Comment:
In context of changing the name from `starts_new` to `does_not_start_new`, the "new" was subtle to me.  Consider adding a human assertion message here to cement why this is expected.

Reply:
Nice suggestion.